docs/articles/viewing-epoch-based-metrics.rst

```diff
@@ -0,0 +1,232 @@
+your script, ``model_def_metrics.py``, contains ``core_context``:
+
+.. code:: python
+
+   if __name__ == "__main__":
+
+     # NEW: Establish new determined.core.Context and pass to main
+     # function.
+     with det.core.init() as core_context:
+         main(core_context=core_context)
+
+Step 2.2: Modify the Train and Validation Methods to Report Epoch-Based Metrics
+===============================================================================
+
+Our script already contains ``core_context.train``. This is used to report training and validation
+metrics.
+
+But, we also want to report epoch-based metrics. To do this, we'll modify the train() method to
+include ``epoch_idx`` as a metric. This allows Determined to keep track of the specific epoch for
+which training loss is being reported:
+
+.. code:: python
+
+   # NEW: Report epoch-based training metrics to Determined
+   # master via core_context.
+   # Index by (batch_idx + 1) * (epoch-1) * len(train_loader)
+   # to continuously plot loss on one graph for consecutive
+   # epochs.
+   core_context.train.report_training_metrics(
+       steps_completed=batches_completed + epoch_idx * len(train_loader),
+       metrics={"train_loss": loss.item(), "epoch": epoch_idx},
+
+   )
+
+Similarly, we'll include ``epoch`` as a metric in the reported validation metrics. This allows
+Determined to track the specific epoch for which the validation loss is being reported:
+
+.. code:: python
+
+   # NEW: Report epoch_based validation metrics to Determined master
+   # via core_context.
+   core_context.train.report_validation_metrics(
+       steps_completed=steps_completed,
+       metrics={"test_loss": test_loss, "epoch": epoch},
+
+   )
+
+Now that we've reported an epoch value, **Epoch** will be an available option for the X-Axis when we
+view our metric data graph in the WebUI.
+
+Step 2.3: Verify the Test Method
+================================
+
+We'll need a ``test()`` function to evaluate the trained model on the test/validation data for the
+current epoch.
+
+Verify your code contains a ``test()`` function header that includes ``args`` and other elements
+needed during the evaluation loop. This function header should pass ``core_context`` into both
+``train()`` and ``test()``:
+
+.. code:: python
+
+   # NEW: Pass args, test_loader, epoch, and steps_completed into
+   # test().
+   test(
+       args,
+       model,
+       device,
+       test_loader,
+       epoch_idx,
+       core_context,
+       steps_completed=steps_completed,
+   )
+   scheduler.step()
+
+In addition, we'll need a ``steps_completed`` variable to plot metrics on a graph in the WebUI. The
+goal is to to track the progress of training by considering both the number of completed training
+batches and the current epoch index. This allows Determined to continuously plot the training loss
+on one graph for consecutive epochs.
+
+Verify your code contains a ``steps_completed`` variable:
```

Comment:
`steps_completed` is a mandatory parameter, so I don't think we need to write docs telling them to verify that it was passed... if they don't pass it, python itself will tell them it's a required parameter.

Reply:
mitigated both comments by removing any steps that are already done in the downloaded scripts.